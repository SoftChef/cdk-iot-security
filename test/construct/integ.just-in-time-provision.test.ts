import * as path from 'path';
import {
  App,
  Stack,
} from 'aws-cdk-lib';
import {
  Template,
} from 'aws-cdk-lib/assertions';
import {
  Runtime,
} from 'aws-cdk-lib/aws-lambda';
import {
  Bucket,
} from 'aws-cdk-lib/aws-s3';
import {
  JustInTimeProvision,
} from '../../src';


const expectedResources: {
  [name: string]: string;
} = {
  caRegistratorPolicy: 'JustInTimeProvisiontestcaseCaRegistratortestcase21FFC435',
  caRegistratorRole: 'JustInTimeProvisiontestcaseCaRegistratortestcaseServiceRoleFB1CB993',
  caRegistrator: 'JustInTimeProvisiontestcaseCaRegistratortestcase93903A27',
  provisionRole: 'JustInTimeProvisiontestcaseProvisionRoleRegistrationConfigRoletestcaseE8C566C2',
  deviceCertificateGenerator: 'JustInTimeProvisiontestcaseDeviceCertificateGeneratortestcase625C5945',
  deviceCertificateGeneratorPolicy: 'JustInTimeProvisiontestcaseDeviceCertificateGeneratortestcasePolicyDeviceCertificateGeneratortestcaseD0261C2C',
  deviceCertificateGeneratorRole: 'JustInTimeProvisiontestcaseDeviceCertificateGeneratortestcaseServiceRoleCA89F5E4',
  verifiersFetcher: 'JustInTimeProvisiontestcaseVerifiersFetchertestcase669A84BA',
  verifiersFetcherRole: 'JustInTimeProvisiontestcaseVerifiersFetchertestcaseServiceRole9B13C41F',
};

const expected: {
  [name: string]: string;
} = {
  lambdaFunctionRuntime: Runtime.NODEJS_14_X.toString(),
  vault: 'another-stack:ExportsOutputRefcaBucketD1A50B2B031F53FA',
};

describe('Integration test', () => {
  test('JustInTimeProvision', () => {
    process.env.BASE_PATH = __dirname;
    process.env.APPS_PATH = path.resolve(__dirname, '..', '..', 'src', 'lambda-assets');
    const app = new App();
    const stack = new Stack(app, 'test-stack');
    const name = 'test-case';
    const anotherStack = new Stack(app, 'another-stack');
    const caBucket = new Bucket(anotherStack, 'caBucket');
    new JustInTimeProvision(stack, name, {
      vault: {
        bucket: caBucket,
        prefix: 'test',
      },
    });
    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
    template.resourceCountIs('AWS::Lambda::Function', 3);
    template.resourceCountIs('AWS::IAM::Role', 4);
    template.resourceCountIs('AWS::IAM::Policy', 4);

    template.hasResourceProperties('AWS::Lambda::Function', {
      Environment: {
        Variables: {
          BUCKET_NAME: {
            'Fn::ImportValue': expected.vault,
          },
          BUCKET_PREFIX: 'test',
          REGISTRATION_CONFIG_ROLE_ARN: {
            'Fn::GetAtt': [
              expectedResources.provisionRole,
              'Arn',
            ],
          },
          VERIFIERS: '"[]"',
        },
      },
      MemorySize: 256,
      Role: {
        'Fn::GetAtt': [
          expectedResources.caRegistratorRole,
          'Arn',
        ],
      },
      Runtime: expected.lambdaFunctionRuntime,
      Timeout: 10,
    });

    template.hasResourceProperties('AWS::Lambda::Function', {
      Environment: {
        Variables: {
          AWS_NODEJS_CONNECTION_REUSE_ENABLED: '1',
          BUCKET_NAME: {
            'Fn::ImportValue': expected.vault,
          },
          BUCKET_PREFIX: 'test',
        },
      },
      Handler: 'index.handler',
      MemorySize: 256,
      Role: {
        'Fn::GetAtt': [
          expectedResources.deviceCertificateGeneratorRole,
          'Arn',
        ],
      },
      Runtime: expected.lambdaFunctionRuntime,
      Timeout: 10,
    });

    template.hasResourceProperties('AWS::Lambda::Function', {
      Environment: {
        Variables: {
          AWS_NODEJS_CONNECTION_REUSE_ENABLED: '1',
          VERIFIERS: '"[]"',
        },
      },
      Handler: 'index.handler',
      Role: {
        'Fn::GetAtt': [
          expectedResources.verifiersFetcherRole,
          'Arn',
        ],
      },
      Runtime: expected.lambdaFunctionRuntime,
    });

    template.hasResourceProperties('AWS::IAM::Policy', {
      PolicyDocument: {
        Statement: [
          {
            Action: [
              'iam:PassRole',
              'iot:RegisterCACertificate',
              'iot:GetRegistrationCode',
              'iot:CreateTopicRule',
              'iot:TagResource',
            ],
            Effect: 'Allow',
            Resource: '*',
          },
        ],
        Version: '2012-10-17',
      },
      PolicyName: expectedResources.caRegistratorPolicy,
      Roles: [
        {
          Ref: expectedResources.caRegistratorRole,
        },
      ],
    });

    template.hasResourceProperties('AWS::IAM::Policy', {
      PolicyDocument: {
        Statement: [
          {
            Action: [
              'iot:DescribeThing',
              'iot:DeleteCertificate',
              'iot:DeleteThing',
              'iot:DescribeCACertificate',
              'iot:ListTagsForResource',
              'iot:DescribeEndpoint',
              'lambda:InvokeFunction',
              'lambda:InvokeAsync',
            ],
            Effect: 'Allow',
            Resource: '*',
          },
        ],
        Version: '2012-10-17',
      },
      PolicyName: expectedResources.deviceCertificateGeneratorPolicy,
      Roles: [
        {
          Ref: expectedResources.deviceCertificateGeneratorRole,
        },
      ],
    });

    template.hasResourceProperties('AWS::IAM::Role', {
      AssumeRolePolicyDocument: {
        Statement: [
          {
            Action: 'sts:AssumeRole',
            Effect: 'Allow',
            Principal: {
              Service: 'iot.amazonaws.com',
            },
          },
        ],
        Version: '2012-10-17',
      },
      ManagedPolicyArns: [
        {
          'Fn::Join': [
            '',
            [
              'arn:',
              {
                Ref: 'AWS::Partition',
              },
              ':iam::aws:policy/service-role/AWSIoTThingsRegistration',
            ],
          ],
        },
      ],
      RoleName: 'ProvisionRoleName-RegistrationConfigRole-test-case',
    });
  });
});
