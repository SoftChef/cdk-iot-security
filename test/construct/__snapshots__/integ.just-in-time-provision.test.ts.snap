// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration test JustInTimeProvision 1`] = `
Object {
  "Parameters": Object {
    "AssetParametersd36f1e961d5ca20ef8a26cbc115022bc46eb9137c0fb18487e55b07dc6c92264ArtifactHash3B5E0809": Object {
      "Description": "Artifact hash for asset \\"d36f1e961d5ca20ef8a26cbc115022bc46eb9137c0fb18487e55b07dc6c92264\\"",
      "Type": "String",
    },
    "AssetParametersd36f1e961d5ca20ef8a26cbc115022bc46eb9137c0fb18487e55b07dc6c92264S3Bucket5EB854A0": Object {
      "Description": "S3 bucket for asset \\"d36f1e961d5ca20ef8a26cbc115022bc46eb9137c0fb18487e55b07dc6c92264\\"",
      "Type": "String",
    },
    "AssetParametersd36f1e961d5ca20ef8a26cbc115022bc46eb9137c0fb18487e55b07dc6c92264S3VersionKey1D21D6F2": Object {
      "Description": "S3 key for asset version \\"d36f1e961d5ca20ef8a26cbc115022bc46eb9137c0fb18487e55b07dc6c92264\\"",
      "Type": "String",
    },
    "AssetParametersd8ccde7eb16f80a322001d51957257632141c8efde61ecbb2592c026f3efa7ebArtifactHashA963BE19": Object {
      "Description": "Artifact hash for asset \\"d8ccde7eb16f80a322001d51957257632141c8efde61ecbb2592c026f3efa7eb\\"",
      "Type": "String",
    },
    "AssetParametersd8ccde7eb16f80a322001d51957257632141c8efde61ecbb2592c026f3efa7ebS3BucketC4EC06DF": Object {
      "Description": "S3 bucket for asset \\"d8ccde7eb16f80a322001d51957257632141c8efde61ecbb2592c026f3efa7eb\\"",
      "Type": "String",
    },
    "AssetParametersd8ccde7eb16f80a322001d51957257632141c8efde61ecbb2592c026f3efa7ebS3VersionKey9D46F397": Object {
      "Description": "S3 key for asset version \\"d8ccde7eb16f80a322001d51957257632141c8efde61ecbb2592c026f3efa7eb\\"",
      "Type": "String",
    },
    "AssetParametersf431e3b1dfccc4e8d1b9d2c0970c1d08dbeaefed64638533de3deb22f025f089ArtifactHash171754A9": Object {
      "Description": "Artifact hash for asset \\"f431e3b1dfccc4e8d1b9d2c0970c1d08dbeaefed64638533de3deb22f025f089\\"",
      "Type": "String",
    },
    "AssetParametersf431e3b1dfccc4e8d1b9d2c0970c1d08dbeaefed64638533de3deb22f025f089S3Bucket293AE0FC": Object {
      "Description": "S3 bucket for asset \\"f431e3b1dfccc4e8d1b9d2c0970c1d08dbeaefed64638533de3deb22f025f089\\"",
      "Type": "String",
    },
    "AssetParametersf431e3b1dfccc4e8d1b9d2c0970c1d08dbeaefed64638533de3deb22f025f089S3VersionKeyB2646B82": Object {
      "Description": "S3 key for asset version \\"f431e3b1dfccc4e8d1b9d2c0970c1d08dbeaefed64638533de3deb22f025f089\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "JustInTimeProvisiontestcaseCaRegistratortestcase21FFC435": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "iam:PassRole",
                "iot:RegisterCACertificate",
                "iot:GetRegistrationCode",
                "iot:CreateTopicRule",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "JustInTimeProvisiontestcaseCaRegistratortestcase21FFC435",
        "Roles": Array [
          Object {
            "Ref": "JustInTimeProvisiontestcaseCaRegistratortestcaseServiceRoleFB1CB993",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "JustInTimeProvisiontestcaseCaRegistratortestcase93903A27": Object {
      "DependsOn": Array [
        "JustInTimeProvisiontestcaseCaRegistratortestcaseServiceRoleDefaultPolicy8C8AE893",
        "JustInTimeProvisiontestcaseCaRegistratortestcaseServiceRoleFB1CB993",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersd36f1e961d5ca20ef8a26cbc115022bc46eb9137c0fb18487e55b07dc6c92264S3Bucket5EB854A0",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersd36f1e961d5ca20ef8a26cbc115022bc46eb9137c0fb18487e55b07dc6c92264S3VersionKey1D21D6F2",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersd36f1e961d5ca20ef8a26cbc115022bc46eb9137c0fb18487e55b07dc6c92264S3VersionKey1D21D6F2",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "BUCKET_NAME": Object {
              "Fn::ImportValue": "another-stack:ExportsOutputRefcaBucketD1A50B2B031F53FA",
            },
            "BUCKET_PREFIX": "test",
            "REGISTRATION_CONFIG_ROLE_ARN": Object {
              "Fn::GetAtt": Array [
                "JustInTimeProvisiontestcaseProvisionRoleRegistrationConfigtestcase7DC6C103",
                "Arn",
              ],
            },
            "VERIFIERS": "\\"[]\\"",
          },
        },
        "Handler": "index.handler",
        "MemorySize": 256,
        "Role": Object {
          "Fn::GetAtt": Array [
            "JustInTimeProvisiontestcaseCaRegistratortestcaseServiceRoleFB1CB993",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "Timeout": 10,
      },
      "Type": "AWS::Lambda::Function",
    },
    "JustInTimeProvisiontestcaseCaRegistratortestcaseServiceRoleDefaultPolicy8C8AE893": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::ImportValue": "another-stack:ExportsOutputFnGetAttcaBucketD1A50B2BArn68FC0213",
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::ImportValue": "another-stack:ExportsOutputFnGetAttcaBucketD1A50B2BArn68FC0213",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "JustInTimeProvisiontestcaseCaRegistratortestcaseServiceRoleDefaultPolicy8C8AE893",
        "Roles": Array [
          Object {
            "Ref": "JustInTimeProvisiontestcaseCaRegistratortestcaseServiceRoleFB1CB993",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "JustInTimeProvisiontestcaseCaRegistratortestcaseServiceRoleFB1CB993": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "JustInTimeProvisiontestcaseDeviceCertificateGeneratortestcase625C5945": Object {
      "DependsOn": Array [
        "JustInTimeProvisiontestcaseDeviceCertificateGeneratortestcaseServiceRoleDefaultPolicyB8C16E44",
        "JustInTimeProvisiontestcaseDeviceCertificateGeneratortestcaseServiceRoleCA89F5E4",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersd8ccde7eb16f80a322001d51957257632141c8efde61ecbb2592c026f3efa7ebS3BucketC4EC06DF",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersd8ccde7eb16f80a322001d51957257632141c8efde61ecbb2592c026f3efa7ebS3VersionKey9D46F397",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersd8ccde7eb16f80a322001d51957257632141c8efde61ecbb2592c026f3efa7ebS3VersionKey9D46F397",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "BUCKET_NAME": Object {
              "Fn::ImportValue": "another-stack:ExportsOutputRefcaBucketD1A50B2B031F53FA",
            },
            "BUCKET_PREFIX": "test",
            "OUTPUT_BUCKET_NAME": Object {
              "Fn::ImportValue": "another-stack:ExportsOutputRefdeviceBucketAD90778AC6CF1AF4",
            },
            "OUTPUT_BUCKET_PREFIX": "",
          },
        },
        "Handler": "index.handler",
        "MemorySize": 256,
        "Role": Object {
          "Fn::GetAtt": Array [
            "JustInTimeProvisiontestcaseDeviceCertificateGeneratortestcaseServiceRoleCA89F5E4",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "Timeout": 10,
      },
      "Type": "AWS::Lambda::Function",
    },
    "JustInTimeProvisiontestcaseDeviceCertificateGeneratortestcasePolicyDeviceCertificateGeneratortestcaseD0261C2C": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "iot:DescribeCACertificate",
                "iot:ListTagsForResource",
                "lambda:InvokeFunction",
                "lambda:InvokeAsync",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "JustInTimeProvisiontestcaseDeviceCertificateGeneratortestcasePolicyDeviceCertificateGeneratortestcaseD0261C2C",
        "Roles": Array [
          Object {
            "Ref": "JustInTimeProvisiontestcaseDeviceCertificateGeneratortestcaseServiceRoleCA89F5E4",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "JustInTimeProvisiontestcaseDeviceCertificateGeneratortestcaseServiceRoleCA89F5E4": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "JustInTimeProvisiontestcaseDeviceCertificateGeneratortestcaseServiceRoleDefaultPolicyB8C16E44": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::ImportValue": "another-stack:ExportsOutputFnGetAttcaBucketD1A50B2BArn68FC0213",
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::ImportValue": "another-stack:ExportsOutputFnGetAttcaBucketD1A50B2BArn68FC0213",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::ImportValue": "another-stack:ExportsOutputFnGetAttdeviceBucketAD90778AArn75A3CD55",
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::ImportValue": "another-stack:ExportsOutputFnGetAttdeviceBucketAD90778AArn75A3CD55",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "JustInTimeProvisiontestcaseDeviceCertificateGeneratortestcaseServiceRoleDefaultPolicyB8C16E44",
        "Roles": Array [
          Object {
            "Ref": "JustInTimeProvisiontestcaseDeviceCertificateGeneratortestcaseServiceRoleCA89F5E4",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "JustInTimeProvisiontestcaseProvisionRoleRegistrationConfigtestcase7DC6C103": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "iot.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSIoTThingsRegistration",
              ],
            ],
          },
        ],
        "RoleName": "ProvisionRoleName-RegistrationConfig-test-case",
      },
      "Type": "AWS::IAM::Role",
    },
    "JustInTimeProvisiontestcaseVerifiersFetchertestcase669A84BA": Object {
      "DependsOn": Array [
        "JustInTimeProvisiontestcaseVerifiersFetchertestcaseServiceRole9B13C41F",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersf431e3b1dfccc4e8d1b9d2c0970c1d08dbeaefed64638533de3deb22f025f089S3Bucket293AE0FC",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersf431e3b1dfccc4e8d1b9d2c0970c1d08dbeaefed64638533de3deb22f025f089S3VersionKeyB2646B82",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersf431e3b1dfccc4e8d1b9d2c0970c1d08dbeaefed64638533de3deb22f025f089S3VersionKeyB2646B82",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "VERIFIERS": "\\"[]\\"",
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "JustInTimeProvisiontestcaseVerifiersFetchertestcaseServiceRole9B13C41F",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "JustInTimeProvisiontestcaseVerifiersFetchertestcaseServiceRole9B13C41F": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
